shell/server/core.js

```diff
@@ -140,17 +141,17 @@ const makeSandstormCore = (grainId) => {
   return new Capnp.Capability(new SandstormCoreImpl(grainId), SandstormCore);
 };
 
-class NotificationHandle {
-  constructor(notificationId, saved, persistentMethods) {
+class NotificationHandle extends PersistentImpl {
+  // TODO(cleanup): Move to a different file.
+
+  constructor(db, saveTemplate, notificationId) {
+    super(db, saveTemplate);
     this.notificationId = notificationId;
-    this.saved = saved;
-    _.extend(this, persistentMethods);
   }
 
   close() {
```

Comment:
The value of `isSaved()` is immutable and determined when the capability is constructed. Whether or not it gets saved, it will call `dismissNotification()` when it is dropped.